### PR TITLE
(PA-4466) Add macOS 12 M1 to build data

### PR DESCRIPTION
**Pull request by mhashizume**
Adds support for macOS 12 Monterey M1/ARM64

### CODE REVIEW

**Comment by mhashizume** [comment]
Once this is merged, how do we assure that _only_ nightlies are built and shipped for this platform? Is it simply a matter of which shipping job is run?

**Review by e-gris — APPROVED** [review]

**Comment by e-gris** [comment]
Here's the task in packaging that does the nightly ship: https://github.com/puppetlabs/packaging/blob/21c64508d74520ccb4951f21791517892a37749d/tasks/jenkins.rake#L297

If the job calls this, then only nightlies are shipped. Moreover, if they are only staged to nightlies, then even should the 'final' version be shipped, they won't appear in 'final' because they were only staged to nightly.

**Comment by mhashizume** [comment]
Thanks for the explanation @e-gris . Going to put a "do not merge" label on this til I can make sure all my ducks are in a row.

**Pull request merged**
